src/core/thread/topology.hpp

```diff
@@ -358,7 +358,10 @@ class Neighbor
      * @returns TRUE if Enhanced Keep-Alive is supported, FALSE otherwise.
      *
      */
-    bool IsEnhancedKeepAliveSupported(void) const { return mVersion >= OPENTHREAD_THREAD_VERSION_1_2; }
+    bool IsEnhancedKeepAliveSupported(void) const
+    {
+        return mState != kStateValid && mVersion >= OPENTHREAD_THREAD_VERSION_1_2;
```

Comment:
Is this supposed to be ``mState != kStateInvalid`` instead of ``mState != kStateValid``?

Reply:
Thank you! This also exposes the test didn't cover resetting timer, so I enhanced the test script to verify the timer is reset.